kmeans clustering with different lines.s_max_pu values

I want to use the line-attribute 's_max_pu' to represent n-1 factor. The grid contains lines from hv- and ehv-level, which have different n-1 factors. This causes an error when I try to apply kmeans clustering: 

`AssertionError: In Bus cluster s_max_pu the values of attribute s_max_pu do not agree:`

I think it would be possible to consider s_max_pu in line aggregation by adding 
`s_max_pu=(l['s_max_pu']* _normed(l['s_nom'])).sum()`
to the aggregatelinegroup function here:
https://github.com/PyPSA/PyPSA/blob/fa63e72dc14969bed48fe480c99a76518d822942/pypsa/networkclustering.py#L182-L198

--- a/doc/release_notes.rst
+++ b/doc/release_notes.rst
@@ -29,6 +29,8 @@ Upcoming Release
 
 * When solving ``n.lopf(pyomo=False)``, PyPSA now constrains the dispatch variables for non extendable components with actual constraints, not with standard variable bounds. This allows retrieving shadow prices for all dispatch variables when running ``n.lopf(pyomo=False, keep_shadowprices=True).   
 
+* Can now cluster lines with different ``s_max_pu`` values.
+
 PyPSA 0.17.0 (23rd March 2020)
 ================================
 

--- a/pypsa/networkclustering.py
+++ b/pypsa/networkclustering.py
@@ -185,6 +185,7 @@ def aggregatelines(network, buses, interlines, line_length_factor=1.0):
             g=(voltage_factor * length_factor * l['g']).sum(),
             b=(voltage_factor * length_factor * l['b']).sum(),
             terrain_factor=l['terrain_factor'].mean(),
+            s_max_pu=(l['s_max_pu'] * _normed(l['s_nom'])).sum(),
             s_nom=l['s_nom'].sum(),
             s_nom_min=l['s_nom_min'].sum(),
             s_nom_max=l['s_nom_max'].sum(),

